bugwarrior/services/clickup.py

```diff
@@ -0,0 +1,150 @@
+import datetime
+import logging
+import typing
+from typing import Generator, Optional
+
+import requests
+
+from bugwarrior import config
+from bugwarrior.services import Client, Issue, Service
+
+log = logging.getLogger(__name__)
+
+
+class ClickupConfig(config.ServiceConfig):
+    service: typing.Literal["clickup"]
+    token: str
+    team: int
+
+    default_priority: str = ""
+
+
+class ClickupClient(Client):
+    """Abstraction of Clickup API v2"""
+
+    def __init__(self, token):
+        self.token = token
+
+    @staticmethod
+    def _get_url_for_tasks(team_id: int, page: int = 0):
+        base_url = "https://api.clickup.com/api/v2/"
+        query = f"include_closed=false&page={page}"
+        return f"{base_url}/team/{team_id}/task?{query}"
+
+    def get_tasks_for_team(self, team_id: int) -> Generator[dict, None, None]:
+        headers = {"Authorization": self.token}
+
+        page = 0
+        while True:
+            response = requests.get(
+                self._get_url_for_tasks(team_id, page), headers=headers
+            )
+            json = self.json_response(response)
+            for task in json["tasks"]:
+                yield task
+
+            if json["last_page"]:
+                break
+
+            # Next page
+            page += 1
+
+
+class ClickupIssue(Issue):
+    ID = "clickupid"
+    DESCRIPTION = "clickupdescription"
+    STATUS = "clickupstatus"
+    UPDATED_AT = "clickupupdated"
+    CREATOR = "clickupcreator"
+    URL = "clickupurl"
+    LIST_NAME = "clickuplistname"
+    PROJECT = "clickupproject"
+    FOLDER = "clickupfolder"
+    SPACE = "clickupspace"
+
+    UDAS = {
+        ID: {"type": "string", "label": "Clickup id"},
+        DESCRIPTION: {"type": "string", "label": "Clickup Description"},
+        STATUS: {"type": "string", "label": "Clickup Status Title"},
+        UPDATED_AT: {"type": "date", "label": "Clickup Updated"},
+        CREATOR: {"type": "string", "label": "Clickup Creator"},
+        URL: {"type": "string", "label": "Clickup URL"},
+        LIST_NAME: {"type": "string", "label": "Clickup List name"},
+        PROJECT: {"type": "string", "label": "Clickup Project id"},
+        FOLDER: {"type": "string", "label": "Clickup Folder id"},
+        SPACE: {"type": "string", "label": "Clickup Space id"},
+    }
+    UNIQUE_KEY = [ID]
```

Comment:
Despite the base class type signature (which is being fixed in #1155), UNIQUE_KEY is supposed to be a tuple.
```suggestion
    UNIQUE_KEY = (ID,)
```